nautobot/core/tests/test_ui.py

```diff
@@ -56,3 +59,93 @@ def test_get_column_headers(self):
             ).get_column_headers(context),
             ["One", "Three"],
         )
+
+
+class BaseTextPanelTest(TestCase):
+    def test_init_set_object_params(self):
+        # Test default settings
+        panel = BaseTextPanel(weight=100)
+        self.assertEqual(panel.render_as, BaseTextPanel.RenderOptions.MARKDOWN)
+        self.assertTrue(panel.render_placeholder)
+
+        # Test initialization with custom arguments
+        panel = BaseTextPanel(weight=100, render_as=BaseTextPanel.RenderOptions.JSON, render_placeholder=False)
+        self.assertEqual(panel.render_as, BaseTextPanel.RenderOptions.JSON)
+        self.assertFalse(panel.render_placeholder)
+
+    @patch.object(Panel, "__init__")
+    def test_init_passes_args_and_kwargs(self, panel_init_mock):
+        custom_template_path = "custom_template_path.html"
+
+        BaseTextPanel(weight=100, body_content_template_path=custom_template_path)
+
+        panel_init_mock.assert_called_once_with(weight=100, body_content_template_path=custom_template_path)
+
+    @patch.object(BaseTextPanel, "get_text")
+    def test_render_body_content(self, get_text_mock):
+        test_cases = [
+            {
+                "render_as": BaseTextPanel.RenderOptions.JSON,
+                "text_content": {"key": "value"},
+                "expected_output": '<code class="language-json">{    &quot;key&quot;: &quot;value&quot;}</code>',
+            },
+            {
+                "render_as": BaseTextPanel.RenderOptions.JSON_PRETTY,
+                "text_content": {"key": "value"},
+                "expected_output": '<pre><code class="language-json">{    &quot;key&quot;: &quot;value&quot;}</code></pre>',
+            },
+            {
+                "render_as": BaseTextPanel.RenderOptions.YAML,
+                "text_content": "key: value",
+                "expected_output": '<pre><code class="language-yaml">&#x27;key: value&#x27;</code></pre>',
```

Comment:
Might be "better" to use the same `{"key": "value"}` text_content here as in the above two cases?